pkg/resource/v1/guestcluster/guestcluster.go

```diff
@@ -0,0 +1,79 @@
+package guestcluster
```

Comment:
AFAIK the stuff in `pkg/resource/v1/` is for common resource implementations. What this here is is a "service" implementation, not a resource. When we want to continue going with the community approach then lets put it into `pkg/`. Otherwise the usual versioned approach would be putting it into something like `service/awsclusterconfig/v1/`, but not in the resource folder.

Reply:
I put it in `pkg/resource/v1` because it will be used by the common resources for all 3 providers. 

Otherwise I would also prefer to have it under `service/`.

Reply:
Ok, I see. The weird thing for me is that we put something that is not a resource into the folder in which the resources are put. That does not add up for me.

Reply:
OK that makes sense. I think it would be clearer if we structured the same way we do under the frameworks. WDYT?


```
pkg
├── cluster
├── label
└── v1
    ├── guestcluster
    ├── key
    └── resource
        ├── certconfig
        ├── chartconfig
        └── encryptionkey
```

Reply:
Lets do that.

Reply:
Sweet. I'll do a separate PR to restructure the existing code.

FYI @tuommaki Let me know if you have any concerns.

Reply:
Done.